### PR TITLE
Implement GET /events/:id  Fixes #27

### LINKED ISSUE #27: Implement `GET /events/:id`
## Tasks:

- Implement `GET /events/:id`
- Add documentation and unit testing.

### PR DESCRIPTION
**Pull request by dilarafirtina**
Closes #27

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -287,6 +287,42 @@
...
+            "required": true,
+            "type": "ObjectId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Events"
+            }
```
Comment (yamanrajab90) [inline swagger.json:310]: Amazing work!  Events definitions were added to the appropriate section.

File: src/controllers/events.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
   return res.status(200).json({ success: true, data: events });
 };
+
+exports.getEvent = async (req, res, next) => {
+  const event = await Event.findById(req.params.id)
+    .select('-pendingApplicants -approvedApplicants -declinedApplicants')
+    .populate('ngo', 'name');
+  if (!event) {
+    return next(new ErrorResponse('No event found', 404));
+  }
```
Comment (yamanrajab90) [inline src/controllers/events.js:18]: Amazing work!  Making use of the ErrorRespons

File: src/controllers/__tests__/events.test.js [hunk 3]
```diff
@@ -95,4 +95,25 @@ describe("Testing events for routes doesn't require auth controls", () => {
         return events;
       });
   });
+
+  it('GET /api/events/:id should retrieve a single event', (done) => {
```
Comment (yamanrajab90) [inline src/controllers/__tests__/events.test.js:99]: Amazing work! all the tests passed

**Review by yamanrajab90 — APPROVED** [review]
This branch is ready to merge into the main.

File: src/controllers/__tests__/events.test.js (hunk 3 above)
Comment (dilarafirtina) [reply src/controllers/__tests__/events.test.js:99]: Thanks a lot for your time @yamanrajab90

**Review by muslimomar — APPROVED** [review]

**Pull request merged**